File: codeforces/101933K-kings-colors/main.py
from math import comb
N, K = map(int, input().split())
ans, sign = 0, 1
for k in range(K, 1, -1):
  ans, sign = (ans + sign*k*(k-1)**(N-1)*comb(K, k)) % (10**9 + 7), -sign
print(ans)

File: codeforces/101933K-kings-colors/main.cpp
#include <iostream>
#define MOD(X) ((X)%(int)(1e9+7))
int main() {
	int N, K, ans = 0;
	std::cin >> N >> K;
	for (int k = K, sign = 1; k >= 2; --k, sign *= -1)
		ans = MOD(ans + MOD());
}
